### DISCUSSION TITLE
Trace log level - docs mistake or other issue?

### DISCUSSION THREAD

**Discussion by tstraley** [flipt-io/flipt#2943] [Q&A] [answered]
https://docs.flipt.io/configuration/overview#logging

These docs say 'trace' is a valid log level that can be configured, but when I attempted to use it, I got this error:

```
2024-04-03T19:49:29Z	WARN	parsing log level, defaulting to INFO	{"level": "trace", "error": "unrecognized level: \"trace\""}
```

Is this just an issue with the docs? Is there some way to enable more detailed logging than debug?

**Comment by markphelps** [comment, marked as answer]
Hey @tstraley ! Thanks for catching this, it is indeed a problem with the docs. Debug is currently the most detailed level of logging we provide. 

We used to use a logging library [logrus](https://pkg.go.dev/github.com/sirupsen/logrus#readme-level-logging) which did have Trace logging.

We've since switched to a better maintained and reportedly faster logging library [zap](https://pkg.go.dev/go.uber.org/zap#hdr-Configuring_Zap) which doesn't have TRACE level.

I'll update the docs to reflect this, thanks again!

Before I close, are you finding the current DEBUG level support lacking?

**Reply by tstraley** [reply]
Thanks @markphelps - I was having one issue that I thought was due to some configuration error that wasn't getting logged as an error and wasn't getting any extra details on the issue with DEBUG, so I tried trace. Turns out that issue was due to not providing my flipt pod with sufficient memory resources and it was crashing before the cobra command would print the reported error - so not actually an issue with insufficient logging, just an issue with getting the relevant logs 😅 

Thanks again
